ledger-service/http-json/src/main/scala/com/digitalasset/http/util/ApiValueToLfValueConverter.scala

```diff
@@ -23,6 +24,6 @@ object ApiValueToLfValueConverter {
     lav1.value.Value => Error \/ lf.value.Value
 
   def apiValueToLfValue: ApiValueToLfValue = { a: lav1.value.Value =>
-    \/.fromEither(NoLoggingValueValidator.validateValue(a)).leftMap(e => Error(e))
+    \/.fromEither(StricterValueValidator.validateValue(a)(NoLogging)).leftMap(e => Error(e))
```

Comment:
This will preserve the original scale of the decimal value. Note that this is applied to both ActiveContract key and payload.